ThrowIT/Utility.m

```diff
@@ -68,5 +70,49 @@ -(void)setAttendanceState:(UIButton *)attendanceButton{
     }
 }
 
++(void)addDistanceDataToList:(NSMutableArray *)partyList fromList:(NSMutableArray *)distanceList{
+    for(int i = 0; i < partyList.count; i++){
+        Party *someParty = partyList[i];
+        someParty.distancesFromUser = distanceList[i];
+    }
+}
+
++(NSMutableArray *)getFilteredListFromList:(NSMutableArray *)partyList withDistanceLimit:(double)distanceLimit withPartyCountlimit:(int)partyCountLimit withRatingLimit:(double) ratingLimit withCompletionHandler:(void (^)(BOOL success))completion{
+    NSMutableArray *filteredList = [NSMutableArray new];
+    NSNumberFormatter *f = [[NSNumberFormatter alloc] init];
+    f.numberStyle = NSNumberFormatterDecimalStyle;
+    for(int i = 0; i < partyList.count; i++){
+        Party *party = partyList[i];
+        //filter rating
+        PFQuery *throwerQuery = [PFQuery queryWithClassName:THROWERCLASS];
+        [throwerQuery whereKey:THROWERKEY equalTo:party.partyThrower];
+        [throwerQuery getFirstObjectInBackgroundWithBlock:^(PFObject * thrower, NSError * error) {
+            filterRunCount++;
+            if(!error){
+                Thrower *partyThrower = (Thrower *)thrower;
+                if(partyThrower.throwerRating >= ratingLimit)
+                    {
+                        //filter distance
+                        if(distanceLimit >= [[f numberFromString:[party.distancesFromUser componentsSeparatedByString:SPACE][0]] doubleValue] || [[party.distancesFromUser componentsSeparatedByString:SPACE][1] isEqualToString:FEET]){
+                            // filter party count
+                            if(party.numberAttending >= partyCountLimit){
+                                [filteredList addObject:party];
+                            }
+                        }
+                    }
+                }
```

Comment:
Use "and" instead of nested ifs
```suggestion
                if(partyThrower.throwerRating >= ratingLimit &&
                   //filter distance
                   (distanceLimit >= [[f numberFromString:[party.distancesFromUser componentsSeparatedByString:SPACE][0]] doubleValue] || [[party.distancesFromUser componentsSeparatedByString:SPACE][1] isEqualToString:FEET]) &&
                    // filter party count
                   party.numberAttending >= partyCountLimit
                    )
                    {
                        [filteredList addObject:party];
                    }
                }
```